File: frontend/src/types/scene.ts
// バックエンドの schemas/scene.py と対応
import type { PaginatedResponse } from './common'

export interface Scene {
  token:       string
  name:        string
  description: string | null
  nbr_samples: number          // サンプル（フレーム）数
  first_sample_token: string
  last_sample_token:  string
  log_token:   string
}

export interface Sample {
  token:       string
  scene_token: string
  timestamp:   number          // UNIX timestamp（マイクロ秒）
  prev:        string | null   // 前フレームのtoken
  next:        string | null   // 次フレームのtoken
}

export type SceneListResponse  = PaginatedResponse<Scene>
export type SampleListResponse = PaginatedResponse<Sample>

// GET /api/v1/logs レスポンスの各エントリ（LogResponse に対応）
export interface Log {
  token:         string
  logfile:       string
  vehicle:       string
  date_captured: string
  location:      string
}

export type LogListResponse = PaginatedResponse<Log>

File: backend/app/schemas/scene.py
from pydantic import BaseModel, ConfigDict


class LogResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    token: str
    logfile: str
    vehicle: str
    date_captured: str
    location: str


class SceneResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    token: str
    log_token: str
    name: str
    description: str | None
    nbr_samples: int
    first_sample_token: str
    last_sample_token: str


class SampleResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    token: str
    scene_token: str
    timestamp: int
    prev: str | None
    next: str | None
